rocketpy/simulation/monte_carlo.py

```diff
@@ -104,6 +110,11 @@ def __init__(
             `out_of_rail_stability_margin`, `out_of_rail_time`,
             `out_of_rail_velocity`, `max_mach_number`, `frontal_surface_wind`,
             `lateral_surface_wind`. Default is None.
+        export_function : callable, optional
+            A function which gets called at the end of a simulation to collect
+            additional data to be exported that isn't pre-defined. Takes the
+            Flight object as an argument and returns a dictionary. Default is None.
```

Comment:
I recommend specifying the dictionary structure. What are the keys and values? Can you give an example to anticipate users' curiosity?

Other thing is that once I read the name `export_function` I automatically imagine a callback function that returns a float, for example:

```python
def mach_number(flight: Flight) -> float:
    return ...

export_function= mach_number
```

Therefore, unless this is only me that has such impression, I believe we could propose a better name for the argument. Maybe `data_collector`or `additional_data_collector`.